src/lib.rs

```diff
@@ -104,6 +106,7 @@ use rand::{
 };
 use serde::{Deserialize, Serialize};
 use std::fmt::{self, Debug, Display, Formatter};
+pub use utils::verify_signature;
```

Comment:
We have a block of public `use` statements above here - probably worth moving this up there.